ocaml/src/alba_client_download.ml

```diff
@@ -34,3 +38,132 @@ let get_object_manifest'
     namespace_id object_name
     lookup_on_nsm_host
     ~consistent_read ~should_cache
+
+
+(* consumers of this method are responsible for freeing
+ * the returned fragment bigstring
+ *)
+let download_fragment
```

Comment:
In hindsight extracting this one from alba_base_client wasn't really needed for this pull request...
(but still it makes alba_base_client thinner, so it's a good thing...)

Besides extracting it I also introduced Prelude.Error here